src/algebra/order/complete_field.lean

```diff
@@ -312,3 +312,27 @@ instance : unique (β ≃+*o γ) := unique_of_subsingleton $ induced_order_ring_
 
 end induced_map
 end linear_ordered_field
+
+section real
+
+/-- Any ring homomorphism `ℝ → ℝ` is an ordered ring isomorphism, and thus there are no
+nontrivial ring homorphism `ℝ → ℝ`. -/
+instance real.ring_hom.unique : unique (ℝ →+* ℝ) :=
+{ default := ring_hom.id ℝ,
+  uniq :=
+  begin
+    intro f,
+    have fmon : monotone f,
+    { intros a b hab,
+      have : f b - f a ≥ 0,
+      { calc
+          f b - f a = f (b - a)                 : (ring_hom.map_sub _ _ _).symm
+            ...     = f (real.sqrt (b - a)^2)   : by rw sq_sqrt (sub_nonneg.mpr hab)
+            ...     = (f (real.sqrt (b - a)))^2 : ring_hom.map_pow _ _ _
+            ...     ≥ 0                         : sq_nonneg _, },
+      linarith, },
+    haveI : subsingleton (ℝ →+*o ℝ):= order_ring_hom.subsingleton,
+    exact congr_arg order_ring_hom.to_ring_hom (subsingleton.elim ⟨f, fmon⟩ ⟨default, monotone_id⟩),
```

Comment:
I think this also works because we have [order_ring_hom.inhabited](https://leanprover-community.github.io/mathlib_docs/algebra/order/hom/ring.html#order_ring_hom.inhabited)?
```suggestion
    exact congr_arg order_ring_hom.to_ring_hom (subsingleton.elim ⟨f, fmon⟩ default),
```

Reply:
Let's see. I am not at my computer so CI will tell us.